qiskit_machine_learning/algorithms/classifiers/pegasos_qsvc.py

```diff
@@ -0,0 +1,285 @@
+            )
+
+        t_0 = datetime.now()
+        y = np.zeros(X.shape[0])
+        for i in range(X.shape[0]):
+            value = self._compute_weighted_kernel_sum(i, X, training=False)
+
+            if value > 0:
+                y[i] = self._label_pos
+            else:
+                y[i] = self._label_neg
+
+        logger.debug("prediction completed after %s", str(datetime.now() - t_0)[:-7])
+
+        return y
+
+    def _compute_weighted_kernel_sum(self, index, X, training) -> float:
+        """Helper function to compute the weighted sum over support vectors used for both training
+        and prediction with the Pegasos algorithm.
+
+        Args:
+            index: fixed index distinguishing some datum
+            X: Features
+            training: flag indicating whether the loop is used within training or prediction
+
+        Returns:
+            float: weighted sum of kernel evaluations employed in the Pegasos algorithm
+        """
+        kernel: Dict[Tuple, np.ndarray] = {}
+        # this value corresponds to a sum of kernel values weighted by their labels and alphas
+        value = 0.0
+        # only loop over the non zero alphas (preliminary support vectors)
+        for j in self._alphas:
+            # perform the kernel evaluations on the fly, as intended
+            if not self._precomputed:
+                # for training
+                if training:
+                    x_j = X[j]
+                # for prediction
+                else:
+                    x_j = self._x_train[j]
+
+                # evaluate kernel function only for the fixed datum and the data with non zero alpha
+                kernel[(index, j)] = kernel.get(
+                    (index, j), self._quantum_kernel.evaluate(X[index], x_j)
+                )
+                kernel_value = kernel[(index, j)]
+
+            # consider a precomputed kernel
+            else:
+                kernel_value = X[index, j]
+
+            value += (
+                # alpha weights the contribution of the associated datum
+                self._alphas[j]
+                # the class membership labels have to be in {-1, +1}
+                * self._label_map[self._y_train[j]]
+                # the offset to the kernel function leads to an implicit bias term
+                * (kernel_value + self._kernel_offset)
+            )
+
+        return value
+
+    @property
+    def quantum_kernel(self) -> QuantumKernel:
+        """Returns quantum kernel"""
+        return self._quantum_kernel
+
+    @quantum_kernel.setter
+    def quantum_kernel(self, quantum_kernel: QuantumKernel):
+        """Sets quantum kernel"""
+        # reset training status
+        self._fit_status = False
+        self._alphas = None
+        self._x_train = None
+        self._n_samples = None
+        self._y_train = None
+        self._label_map = None
+        self._label_pos = None
+        self._label_neg = None
+
+        self._quantum_kernel = quantum_kernel
```

Comment:
Should we have other getter/setters for the other contructors params so for instance I can get/change `num_steps`. What about switching to a precomputed kernel - might be easier with my suggestion above rather than the separate params. Since C is a parameter of the parent  SVC class I think I can set/get that already by virtue of the parent

Reply:
Added getters/setters and more tests.